{"name": "test_case[case_data0]", "status": "broken", "statusDetails": {"message": "TypeError: a bytes-like object is required, not 'NoneType'", "trace": "self = <test_configs.TestConfigs object at 0x115baa5f8>\ncase_data = {'case_id': 1, 'check': {'check_type': 'Regular_check', 'expected_code': 200, 'expected_result': ['\"ret\": 0']}, 'global_relevance': None, 'info': None, ...}\nsetupClass = {}\n\n    @pytest.mark.skipif(case_dict[\"test_info\"][\"product_version\"] in product_version,\n                        reason=\"该用例所属版本为：{0}，在本次排除版本{1}内\".format(case_dict[\"test_info\"][\"product_version\"],\n                                                                 product_version))\n    @pytest.mark.skipif(case_dict[\"test_info\"][\"case_level\"] not in case_level,\n                        reason=\"该用例的用例等级为：{0}，不在本次运行级别{1}内\".format(case_dict[\"test_info\"][\"case_level\"], case_level))\n    @pytest.mark.run(order=case_dict[\"test_info\"][\"run_order\"])\n    @allure.severity(case_dict[\"test_info\"][\"case_level\"])\n    @pytest.mark.parametrize(\"case_data\", case_dict[\"test_case\"], ids=[])\n    @allure.story(\"configs\")\n    def test_case(self, case_data, setupClass):\n        self.relevance = setupTest(relevance_path, case_data, setupClass)\n>       api_send_check(case_data, case_dict, case_path, self.relevance)\n\n/Users/nali/AutoApiTest/tests/TestScript/configsAPI/test_configs.py:47: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n/Users/nali/AutoApiTest/common/unit/api_send_check.py:26: in api_send_check\n    res = api_send.api_send(case_data, case_dict, case_path, relevance)\n/Users/nali/AutoApiTest/common/unit/api_send.py:117: in api_send\n    allure.attach(case_dict[\"info\"], \"用例描述：\")\n/Users/nali/anaconda3/lib/python3.7/site-packages/allure_commons/_allure.py:176: in __call__\n    plugin_manager.hook.attach_data(body=body, name=name, attachment_type=attachment_type, extension=extension)\n/Users/nali/anaconda3/lib/python3.7/site-packages/pluggy/hooks.py:286: in __call__\n    return self._hookexec(self, self.get_hookimpls(), kwargs)\n/Users/nali/anaconda3/lib/python3.7/site-packages/pluggy/manager.py:93: in _hookexec\n    return self._inner_hookexec(hook, methods, kwargs)\n/Users/nali/anaconda3/lib/python3.7/site-packages/pluggy/manager.py:87: in <lambda>\n    firstresult=hook.spec.opts.get(\"firstresult\") if hook.spec else False,\n/Users/nali/anaconda3/lib/python3.7/site-packages/allure_pytest/listener.py:220: in attach_data\n    self.allure_logger.attach_data(uuid4(), body, name=name, attachment_type=attachment_type, extension=extension)\n/Users/nali/anaconda3/lib/python3.7/site-packages/allure_commons/reporter.py:117: in attach_data\n    plugin_manager.hook.report_attached_data(body=body, file_name=file_name)\n/Users/nali/anaconda3/lib/python3.7/site-packages/pluggy/hooks.py:286: in __call__\n    return self._hookexec(self, self.get_hookimpls(), kwargs)\n/Users/nali/anaconda3/lib/python3.7/site-packages/pluggy/manager.py:93: in _hookexec\n    return self._inner_hookexec(hook, methods, kwargs)\n/Users/nali/anaconda3/lib/python3.7/site-packages/pluggy/manager.py:87: in <lambda>\n    firstresult=hook.spec.opts.get(\"firstresult\") if hook.spec else False,\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <allure_commons.logger.AllureFileLogger object at 0x115af7d30>\nbody = None\nfile_name = '822ed219-a162-4291-9766-b65a661d81de-attachment.attach'\n\n    @hookimpl\n    def report_attached_data(self, body, file_name):\n        destination = os.path.join(self._report_dir, file_name)\n        with open(destination, 'wb') as attached_file:\n            if isinstance(body, text_type):\n                attached_file.write(body.encode('utf-8'))\n            else:\n>               attached_file.write(body)\nE               TypeError: a bytes-like object is required, not 'NoneType'\n\n/Users/nali/anaconda3/lib/python3.7/site-packages/allure_commons/logger.py:62: TypeError"}, "steps": [{"name": "get请求接口", "status": "broken", "statusDetails": {"message": "TypeError: a bytes-like object is required, not 'NoneType'\n", "trace": "  File \"/Users/nali/AutoApiTest/common/unit/api_send.py\", line 117, in api_send\n    allure.attach(case_dict[\"info\"], \"用例描述：\")\n  File \"/Users/nali/anaconda3/lib/python3.7/site-packages/allure_commons/_allure.py\", line 176, in __call__\n    plugin_manager.hook.attach_data(body=body, name=name, attachment_type=attachment_type, extension=extension)\n  File \"/Users/nali/anaconda3/lib/python3.7/site-packages/pluggy/hooks.py\", line 286, in __call__\n    return self._hookexec(self, self.get_hookimpls(), kwargs)\n  File \"/Users/nali/anaconda3/lib/python3.7/site-packages/pluggy/manager.py\", line 93, in _hookexec\n    return self._inner_hookexec(hook, methods, kwargs)\n  File \"/Users/nali/anaconda3/lib/python3.7/site-packages/pluggy/manager.py\", line 87, in <lambda>\n    firstresult=hook.spec.opts.get(\"firstresult\") if hook.spec else False,\n  File \"/Users/nali/anaconda3/lib/python3.7/site-packages/pluggy/callers.py\", line 208, in _multicall\n    return outcome.get_result()\n  File \"/Users/nali/anaconda3/lib/python3.7/site-packages/pluggy/callers.py\", line 80, in get_result\n    raise ex[1].with_traceback(ex[2])\n  File \"/Users/nali/anaconda3/lib/python3.7/site-packages/pluggy/callers.py\", line 187, in _multicall\n    res = hook_impl.function(*args)\n  File \"/Users/nali/anaconda3/lib/python3.7/site-packages/allure_pytest/listener.py\", line 220, in attach_data\n    self.allure_logger.attach_data(uuid4(), body, name=name, attachment_type=attachment_type, extension=extension)\n  File \"/Users/nali/anaconda3/lib/python3.7/site-packages/allure_commons/reporter.py\", line 117, in attach_data\n    plugin_manager.hook.report_attached_data(body=body, file_name=file_name)\n  File \"/Users/nali/anaconda3/lib/python3.7/site-packages/pluggy/hooks.py\", line 286, in __call__\n    return self._hookexec(self, self.get_hookimpls(), kwargs)\n  File \"/Users/nali/anaconda3/lib/python3.7/site-packages/pluggy/manager.py\", line 93, in _hookexec\n    return self._inner_hookexec(hook, methods, kwargs)\n  File \"/Users/nali/anaconda3/lib/python3.7/site-packages/pluggy/manager.py\", line 87, in <lambda>\n    firstresult=hook.spec.opts.get(\"firstresult\") if hook.spec else False,\n  File \"/Users/nali/anaconda3/lib/python3.7/site-packages/pluggy/callers.py\", line 208, in _multicall\n    return outcome.get_result()\n  File \"/Users/nali/anaconda3/lib/python3.7/site-packages/pluggy/callers.py\", line 80, in get_result\n    raise ex[1].with_traceback(ex[2])\n  File \"/Users/nali/anaconda3/lib/python3.7/site-packages/pluggy/callers.py\", line 187, in _multicall\n    res = hook_impl.function(*args)\n  File \"/Users/nali/anaconda3/lib/python3.7/site-packages/allure_commons/logger.py\", line 62, in report_attached_data\n    attached_file.write(body)\n"}, "attachments": [{"name": "请求接口：", "source": "a59cff9c-573a-499c-9f43-6630254d2532-attachment.attach"}, {"name": "用例描述：", "source": "822ed219-a162-4291-9766-b65a661d81de-attachment.attach"}], "start": 1617608092644, "stop": 1617608092645}], "attachments": [{"name": "log", "source": "a8379b88-5e4a-49d1-8e5b-b95cc3668383-attachment.txt", "type": "text/plain"}, {"name": "stdout", "source": "4a07596a-e173-4839-a540-074770de2aab-attachment.txt", "type": "text/plain"}], "parameters": [{"name": "case_data", "value": "{'test_name': 'configs_1', 'case_id': 1, 'is_run': None, 'info': None, 'parameter': None, 'check': {'expected_result': ['\"ret\": 0'], 'check_type': 'Regular_check', 'expected_code': 200}, 'global_relevance': None, 'relevance': None}"}], "start": 1617608092641, "stop": 1617608092650, "uuid": "03ebcb58-f0e7-458e-80e5-80d19c247c73", "historyId": "d8d925246c1fec44edb81b2b419d34e8", "testCaseId": "31717c13dc002ec64eea43528e4bf521", "fullName": "tests.TestScript.configsAPI.test_configs.TestConfigs#test_case", "labels": [{"name": "severity", "value": "blocker"}, {"name": "feature", "value": "configs"}, {"name": "story", "value": "configs"}, {"name": "tag", "value": "@pytest.mark.skipif(False, reason='该用例的用例等级为：blocker，不在本次运行级别['blocker', 'critical', 'normal', 'minor']内')"}, {"name": "tag", "value": "run(order=1)"}, {"name": "parentSuite", "value": "tests.TestScript.configsAPI"}, {"name": "suite", "value": "test_configs"}, {"name": "subSuite", "value": "TestConfigs"}, {"name": "host", "value": "anqidediannao.local"}, {"name": "thread", "value": "13099-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.TestScript.configsAPI.test_configs"}]}